### PR TITLE
chore: [CDS-80523]: yaml input component for default object type

### PR DESCRIPTION
**Pull request by PratyushGar**
### Summary

- Adds YAML editor input for object data type fallback
- For multiple dependencies default to textArea component

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/70-pipeline/y1/components/InputsForm/utils.ts [hunk 1]
```diff
@@ -67,7 +68,8 @@ function processMetadata(
     }
   })
 
-  const metadata: UIInputMetadata = allMetadata.length === 1 ? allMetadata[0] : { ...defaultInputMetadata }
+  const metadata: UIInputMetadata =
```
Comment (srdjan-harness) [inline src/modules/70-pipeline/y1/components/InputsForm/utils.ts:71]: This means that we will use text_area (multiFieldPropertyDefaultInputMetadata) if input is used in more than 1 place.

**Review by srdjan-harness — APPROVED** [review]

File: src/modules/70-pipeline/y1/components/InputsForm/utils.ts (hunk 1 above)
Comment (PratyushGar) [reply src/modules/70-pipeline/y1/components/InputsForm/utils.ts:71]: Yes, the idea is if multiple references are there, textarea would help input a script, string, boolean, yaml, etc.

File: src/modules/70-pipeline/y1/components/InputFactory/Inputs/ObjectInput.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      skipRenderValueInExpressionLabel
+      expressionRender={() => editor}
+    >
+      {editor}
+    </MultiTypeFieldSelector>
+  )
+}
+
+export class ObjectInput extends InputComponent<InputsFormValues> {
+  public internalType = PrimitiveInputType.string
```
Comment (rameshx) [inline src/modules/70-pipeline/y1/components/InputFactory/Inputs/ObjectInput.tsx:61]: Shouldn't this be `PrimitiveInputType.object`?

Comment (PratyushGar) [reply src/modules/70-pipeline/y1/components/InputFactory/Inputs/ObjectInput.tsx]: Yes, done. Thanks!

**Review by rameshx — APPROVED** [review]

**Pull request merged**